aggregate-service/src/main/java/org/opentestsystem/rdw/olap/service/impl/DefaultOrganizationSearchService.java

```
@@ -0,0 +1,71 @@
+package org.opentestsystem.rdw.olap.service.impl;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import org.opentestsystem.rdw.olap.service.OrganizationSearchService;
+import org.opentestsystem.rdw.olap.service.OrganizationService;
+import org.opentestsystem.rdw.reporting.common.model.Organization;
+import org.opentestsystem.rdw.reporting.common.model.OrganizationType;
+import org.opentestsystem.rdw.reporting.common.model.School;
+import org.opentestsystem.rdw.reporting.common.web.security.User;
+import org.opentestsystem.rdw.security.PermissionScope;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static org.opentestsystem.rdw.olap.model.AggregateServicePermission.AggregateRead;
+
+@Service
+class DefaultOrganizationSearchService implements OrganizationSearchService {
+
+    private final OrganizationService service;
+
+    @Autowired
+    DefaultOrganizationSearchService(final OrganizationService service) {
+        this.service = service;
+    }
+
+    @Override
+    public Set<Organization> getOrganizations(final User user, final String nameSearch) {
+
+        // fail fast
+        if (Strings.isNullOrEmpty(nameSearch)) {
+            return ImmutableSet.of();
+        }
+
+        final PermissionScope permissionScope = user.getPermissionScopeByPermissionId(AggregateRead);
+
+        final Predicate<Organization> securityFilter = permissionScope.isStatewide()
```

Not sure why but i find this hard to read. I think a ternary operator returning different lambdas, all in-line is just too much. Perhaps break out the filter into a separate class (can be static in this module to avoid new module). And perhaps have one filter that just exits early if the permission scope is statewide? Although i guess we could have AlwaysTrue predicate.

---

The always true predicate was technically more efficient than having one predicate with a fail fast hence the choice to use it - i can go ahead and put this in a class.